memgpt/memory.py

```diff
@@ -700,17 +727,23 @@ def save(self):
         # don't need to save data source, since we assume data source data is already loaded into the agent index
         utils.save_agent_index(self.index, self.agent_config)
 
-    def insert(self, memory_string):
-        self.index.insert(memory_string)
+    def insert(self, memory_string: str, embedding: Optional[VectorEmbedding] = None):
+        model_name = self.agent_config.embedding_model
+        metadata = {
+            "timestamp": get_local_time(),
+            **({"model_name": model_name} if self.model_name else {}),
+        }
+        document = Document(text=memory_string, embedding=embedding, metadata=metadata)
+        self.index.insert(document)
 
         # TODO: figure out if this needs to be refreshed (probably not)
         self.retriever = VectorIndexRetriever(
             index=self.index,
             similarity_top_k=self.top_k,
         )
 
-    async def a_insert(self, memory_string):
-        return self.insert(memory_string)
+    async def a_insert(self, memory_string: str, embedding: Optional[VectorEmbedding] = None):

```

Comment:
Passing in the embeddings into `a_insert` was actually a bug in the code (will be fixed by #316), so `a_insert` and `insert` should actually only take in a memory string. The `ArchivalMemory` class should be responsible for computing the embeddings and any additional metadata, since this may depend on the underlying storage being used.

Reply:
I thnk I've dealt with this, but I need to re-test. I've set up a docker compose setup with containerized postgres & memgpt that I'll do a separate PR for. I plan to leverage this for some automated tests. Everything appeared to be working before I did a merge yesterday, so I'm optimistic.

(I'll include my tests for inspection, but w/o the docker stuff. That's going to warrant some design discussion to be sure we're on the same page, consider use cases I may have missed, etc.)

I'm leaving this comment unresolved until I'm satisfied with my tests. I hope to be ready by tomorrow.

Reply:
Sounds good, let me know when I should re-review!